packages/@lwc/ssr-runtime/src/render.ts

```diff
@@ -27,6 +27,25 @@ export function* renderAttrs(instance: LightningElement, attrs: Attributes) {
     yield mutationTracker.renderMutatedAttrs(instance);
 }
 
+export function renderAttrsNoYield(
+    emit: (segment: string) => void,
+    instance: LightningElement,
+    attrs: Attributes
+) {
+    if (!attrs) {
+        return;
+    }
+    for (const attrName of Object.getOwnPropertyNames(attrs)) {
+        const attrVal = attrs[attrName];
+        if (typeof attrVal === 'string') {
+            emit(attrVal === '' ? ` ${attrName}` : ` ${attrName}="${escapeAttrVal(attrVal)}"`);
+        } else if (attrVal === null) {
+            emit('');
+        }
+    }
+    emit(mutationTracker.renderMutatedAttrs(instance));
+}
+
```

Comment:
I made a note of it in the comments & PR description, but also calling it out here. This newly-added function (as well as `fallbackTmplNoYield` below) are the non-generator alternatives of the existing functions that are exported. If the number of helper functions grows beyond two, or if the complexity of these functions grows, we may want to reconsider how to approach this. This approach seemed to okay to me for now, but I'm open to other alternatives.